src/hacbs/components/ApplicationDetails/tabs/overview/visualization/hooks/useAppWorkflowData.ts

```diff
@@ -1,129 +1,582 @@
+      applicationTestNodes,
+      buildTasks,
+      componentIntegrationTestTasks,
+      componentIntegrationTests,
+      componentTestNodes,
+      expanded,
+    ],
+  );
 
-  const nodes: WorkflowNode[] = workflowToNodes(workflowObject);
+  const staticEnvironments = environments.filter(
+    (e) => getEnvironmentType(e) === EnvironmentType.static,
+  );
+
+  const staticEnvironmentNodes = React.useMemo(() => {
+    const nodes = staticEnvironments.length
+      ? staticEnvironments.map((staticEnvironment) => {
+          const prevEnv = staticEnvironments.find(
+            (e) => e.metadata.name === staticEnvironment.spec.parentEnvironment,
+          );
+          return resourceToPipelineNode(
+            staticEnvironment,
+            WorkflowNodeType.STATIC_ENVIRONMENT,
+            prevEnv ? [prevEnv.metadata.uid] : applicationIntegrationTestTasks,
+            healthStatus,
+          );
+        })
+      : [
+          emptyPipelineNode(
+            'no-static-environments',
+            'Static environments',
+            WorkflowNodeType.STATIC_ENVIRONMENT,
+            applicationIntegrationTestTasks,
+          ),
+        ];
+    const maxWidth = nodes.reduce((max, node) => Math.max(max, node.width), 0);
+    nodes.forEach((n) => (n.width = maxWidth));
+    return nodes;
+  }, [applicationIntegrationTestTasks, healthStatus, staticEnvironments]);
+
+  const staticEnvironmentGroup = React.useMemo(
+    () =>
+      allResourcesLoaded && testsGroup?.id
+        ? groupToPipelineNode(
+            'static-environments',
+            'Static environments',
+            WorkflowNodeType.STATIC_ENVIRONMENT,
+            expanded ? applicationIntegrationTestTasks : [testsGroup.id],
+            expanded,
+            expanded ? staticEnvironmentNodes?.map((c) => c.id) : undefined,
+            staticEnvironmentNodes,
+            staticEnvironments,
+            healthStatus,
+          )
+        : undefined,
+    [
+      allResourcesLoaded,
+      applicationIntegrationTestTasks,
+      expanded,
+      healthStatus,
+      staticEnvironmentNodes,
+      staticEnvironments,
+      testsGroup?.id,
+    ],
+  );
+
+  const lastStaticEnv = getLastEnvironments(staticEnvironments);
+
+  const releasePlanNodes = React.useMemo(() => {
+    const nodes = releasePlans.length
+      ? releasePlans.map((releasePlan) =>
+          resourceToPipelineNode(
+            releasePlan,
+            WorkflowNodeType.RELEASE,
+            lastStaticEnv,
+            pipelineRunStatus(releasePlan),
+          ),
+        )
+      : [
+          emptyPipelineNode(
+            'no-release-plans',
+            'Release plans',
+            WorkflowNodeType.RELEASE,
+            lastStaticEnv,
+          ),
+        ];
+    const maxWidth = nodes.reduce((max, node) => Math.max(max, node.width), 0);
+    nodes.forEach((n) => (n.width = maxWidth));
+    return nodes;
+  }, [lastStaticEnv, releasePlans]);
+
+  const releasePlanGroup = React.useMemo(
```

Comment:
Release bubble should be backed by `Release` CR not `ReleasePlan`. ReleasePlan CR should be the source for managed environment bubble. 
3rd and 4th answer in the [working with commits](https://docs.google.com/document/d/1gqvogvMDH9xyUOaxXg5AHTaJTstvTJtmqC77vHKZeUA/edit?hl=en#heading=h.9u1mu59jwbzs) doc covers this point